### PR TITLE
Comment Detail: Create the initial view 

### PR DESCRIPTION
**Pull request by dvdchr**
Refs #17087

## Summary

This adds the initial view for comment details, along with the feature flag. Some notes for review:

- The feature flag is wired in My Site > Comments, so you don't have to manually change the code.
- The new view is still empty, since more PRs are coming to add on this one. But I've wired in the "Edit" button to connect to the new `EditCommentTableViewController` component developed in #17000

#### Scenario 1: Ensure flow correctness with flag disabled

- Go to My Site > Comments, and tap one of the comments.
- Ensure that the previous comment details screen is shown and works as expected.

#### Scenario 2: The new detail screen should be reachable with flag enabled.

- Enable the `New Comment Details` feature flag.
- Go to My Site > Comments. Tap one of the comments.
- Verify that the (empty) new view is shown 😅 
- Verify that the Edit button correctly shows the new Edit Comment screen (although it's still incomplete, progress for this is tracked in #17000).

## Regression Notes
1. Potential unintended areas of impact
Current comment details is not shown correctly when tapped from My Site > Comments.

2. What I did to test those areas of impact (or what existing automated tests I relied on)
Manually tested that the implementation is correct.

3. What automated tests I added (or what prevented me from doing so)
n/a, the feature is behind a disabled flag.

PR submission checklist:

- [x] I have completed the Regression Notes.
- [x] I have considered adding unit tests for my changes.
- [x] I have considered adding accessibility improvements for my changes.
- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WordPress/Classes/ViewRelated/Comments/CommentDetailViewController.swift [hunk 1]
```diff
@@ -0,0 +1,76 @@
...
+            self.tableView.reloadData()
+        }
+    }
+
+}
+
+// MARK: - Localization
+
+private extension String {
+    static let editBarButtonTitle = NSLocalizedString("Edit", comment: "Edits a Comment")
```
Comment (ScoutHarris) [inline WordPress/Classes/ViewRelated/Comments/CommentDetailViewController.swift]: Is this needed?

File: WordPress/Classes/ViewRelated/Comments/CommentsViewController.m [hunk 2]
```diff
@@ -263,13 +263,20 @@ - (void)tableView:(UITableView *)tableView didSelectRowAtIndexPath:(NSIndexPath
...
         return;
     }
-    
+
     // At last, push the details
-    Comment *comment            = [self.tableViewHandler.resultsController objectAtIndexPath:indexPath];
-    CommentViewController *vc   = [CommentViewController new];
-    vc.comment                  = comment;
+    Comment *comment = [self.tableViewHandler.resultsController objectAtIndexPath:indexPath];
+    UIViewController *detailViewController;
```
Comment (ScoutHarris) [inline WordPress/Classes/ViewRelated/Comments/CommentsViewController.m:269]: 👍

**Review by ScoutHarris — APPROVED** [review]
Hey @dvdchr . Just a question about a bit of code that looks unused. Otherwise 👍 !

:shipit:

File: WordPress/Classes/ViewRelated/Comments/CommentDetailViewController.swift (hunk 1 above)
Comment (dvdchr) [reply WordPress/Classes/ViewRelated/Comments/CommentDetailViewController.swift]: Nope. I forgot to clean it up. Thanks for noticing this! 😄 . Removed in https://github.com/wordpress-mobile/WordPress-iOS/pull/17088/commits/ff524a72e625a85a46649dd59d2df0e562928fd6.

**Pull request merged**